torch/utils/data/_utils/worker.py

```diff
@@ -75,8 +125,37 @@ def _worker_loop(dataset, index_queue, data_queue, done_event, collate_fn, seed,
 
         data_queue.cancel_join_thread()
 
-        if init_fn is not None:
-            init_fn(worker_id)
+        global _worker_info
+        _worker_info = WorkerInfo(id=worker_id, num_workers=num_workers,
+                                  seed=seed, dataset=dataset)
+
+        from torch.utils.data import _DataLoaderStrategy
+
+        init_exception = None
+
+        try:
+            if init_fn is not None:
+                init_fn(worker_id)
+
+            if mode == _DataLoaderStrategy.Iterable:
+                dataset_iter = iter(dataset)
+
+        except Exception:
```

Comment:
Does this try, except just need to be around init_fn?

Reply:
And `iter`